drivers/include/mma7660.h

```diff
@@ -0,0 +1,200 @@
+#define MMA7660_INTSOURCE_UDLR        0x02
+#define MMA7660_INTSOURCE_TAP         0x04
+#define MMA7660_INTSOURCE_AUTOSLEEP   0x08
+#define MMA7660_INTSOURCE_MEASURE     0x10
+#define MMA7660_INTSOURCE_SHX         0x20
+#define MMA7660_INTSOURCE_SHY         0x40
+#define MMA7660_INTSOURCE_SHZ         0x80
+
+#define MMA7660_SR_AMPD   0x00
+#define MMA7660_SR_AM64   0x01
+#define MMA7660_SR_AM32   0x02
+#define MMA7660_SR_AM16   0x03
+#define MMA7660_SR_AM8    0x04
+#define MMA7660_SR_AM4    0x05
+#define MMA7660_SR_AM2    0x06
+#define MMA7660_SR_AM1    0x07
+#define MMA7660_SR_AW32   0x00
+#define MMA7660_SR_AW16   0x08
+#define MMA7660_SR_AW8    0x10
+#define MMA7660_SR_AW1    0x18
+#define MMA7660_PDET_X 0x20
+#define MMA7660_PDET_Y 0x40
+#define MMA7660_PDET_Z 0x80
+
+#define MMA7660_ADDR    0x4C
+
+/**
+ * @brief   Initialize an MMA7660 device
+ *
+ * @param[out] dev          device descriptor
+ * @param[in] i2c           I2C bus the device is connected to
+ * @param[in] address       I2C address of the device, generally 0x4C
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_init(mma7660_t *dev, i2c_t i2c, uint8_t address);
+
+
+/**
+ * @brief   Set the mode register
+ *
+ * @param[in] dev          device descriptor
+ *
+ * See page 17 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for information about the values of the parameters
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_set_mode(mma7660_t *dev, uint8_t active,
+    uint8_t autowake, uint8_t autosleep, uint8_t prescale);
+
+/**
+ * @brief   Read the tilt register
+ *
+ * @param[in] dev          device descriptor
+ * @param[out] res         tilt register contents
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_read_tilt(mma7660_t *dev, uint8_t *res);
+
+/**
+ * @brief   Write the sleep count register
+ *
+ * @param[in] dev          device descriptor
+ * @param[in] sleep        sleep count
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_write_sleep_count(mma7660_t *dev, uint8_t sleep);
+
+/**
+ * @brief   Configure the interrupt sources
+ *
+ * @param[in] dev             device descriptor
+ * @param[in] isource_flags   interrupt source flags
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_interrupts(mma7660_t *dev, uint8_t isource_flags);
+
+/**
+ * @brief   Configure the sample rate
+ *
+ * @param[in] dev             device descriptor
+ *
+ * See page 18 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for details about the parameters
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_samplerate(mma7660_t *dev, uint8_t amsr, uint8_t awsr, uint8_t filt);
```

Comment:
IMHO, this should also be configurable during initalization through the `mma7660_params_t` struct.